src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,379 @@
+namespace NumericalFluxes {
+
+/*!
+ * \brief The internal penalty flux for first-order elliptic equations.
+ *
+ * \details Computes the internal penalty numerical flux (see e.g.
+ * \cite HesthavenWarburton, section 7.2) dotted with the interface unit normal.
+ *
+ * We implement here a suggested generalization of the internal penalty flux
+ * for any set of elliptic PDEs. It is designed for fluxes (i.e. principal parts
+ * of the PDEs) that may depend on the dynamic fields. This is the case for the
+ * velocity potential equation of binary neutron stars, for example.
+ *
+ * Formulating the elliptic PDEs in terms of a first-order _auxiliary_ variable
+ * \f$v\f$ for each _primal_ variable \f$u\f$, they take the _flux-form_
+ *
+ * \f{align}
+ * -\partial_i F^i_A + S_A = f_A
+ * \f}
+ *
+ * where the fluxes and sources are indexed by the variables and we have
+ * defined \f$f_A\f$ as those sources that are independent of the variables.
+ * We now choose the internal penalty numerical fluxes \f$F^{*i}\f$ as follows
+ * for each variable \f$u\f$ and their corresponding auxiliary variable \f$v\f$:
+ *
+ * \f{align}
+ * n_i F^{*i}_u &= \frac{1}{2} n_i \left( F^i_u(\partial_j
+ * F^j_v(u^\mathrm{int})) + F^i_u(\partial_j F^j_v(u^\mathrm{ext}))
+ * \right) - \sigma n_i \left(F^i_u(n\otimes u^\mathrm{int}) - F^i_u(
+ * n\otimes u^\mathrm{ext})\right) \\
+ * n_i F^{*i}_v &= \frac{1}{2} n_i \left(F^i_v(u^\mathrm{int}) +
+ * F^i_v(u^\mathrm{ext})\right)
+ * \f}
+ *
+ * Note that we have assumed \f$n^\mathrm{ext}_i=-n_i\f$ here, i.e. face normals
+ * don't depend on the dynamic variables (which may be discontinuous on element
+ * faces). This is the case for the problems we are expecting to solve, because
+ * those will be on fixed background metrics (e.g. a conformal metric for the
+ * XCTS system).
+ *
+ * Also note that the numerical fluxes intentionally don't depend on the
+ * auxiliary field values \f$v\f$. This property allows us to use the numerical
+ * fluxes also for the second-order (or _primal_) DG formulation, where we
+ * remove the need for an auxiliary variable. For the first-order system we
+ * could replace the divergence in \f$F^{*i}_u\f$ with \f$v\f$, which would
+ * result in a generalized _stabilized central flux_. It resembles the
+ * Local-Lax-Friedrichs flux for evolution systems but remains consistent in the
+ * stabilizer term even for principal parts that depend on dynamic variables.
+ *
+ * For a Poisson system \f$-\Delta u(x) = f(x)\f$ with auxiliary variable
+ * \f$v_i=\partial_i u\f$ this numeric flux reduces to the standard internal
+ * penalty flux (see e.g. \cite HesthavenWarburton, section 7.2, or
+ * \cite Arnold2002)
+ *
+ * \f{align}
+ * n_i F^{*i}_u = n_i v_i^* = \frac{1}{2} n_i \left(\partial_i u^\mathrm{int} +
+ * \partial_i u^\mathrm{ext}\right) - \sigma \left(u^\mathrm{int} -
+ * u^\mathrm{ext}\right)
+ * n_i F^{*i}_{v_j} = n_j u^* = \frac{1}{2} n_j \left(u^\mathrm{int} +
+ * u^\mathrm{ext}\right)
+ * \f}
+ *
```

Comment:
I don't understand this equation, because the indices don't appear to match. And where does \sigma go in the right-most terms?

Reply:
What confuses you? The indices match I think. The index `i` is contracted and the lower index `j` is free. On the LHS I keep the index at the `v` because that's where it comes from. The `sigma` only appears in the numerical flux for `u`.

Reply:
The major confusion is that this renders as a single equation instead of two

Reply:
oh, fixed that. does it makes sense now?